### PR TITLE
Add dot-all regex support to preset-env and standalone

### PR DESCRIPTION
**Pull request by existentialism**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | 
| Patch: Bug Fix?          | n
| Major: Breaking Change?  | n
| Minor: New Feature?      | y
| Tests Added + Pass?      | y/y
| Documentation PR         | 
| Any Dependency Changes?  |
| License                  | MIT

Follow up to https://github.com/babel/babel/pull/7059

### CODE REVIEW

File: packages/babel-preset-env/src/debug.js [hunk 1]
```diff
@@ -15,9 +15,23 @@ export const logMessage = (message, context) => {
 export const logPlugin = (plugin, targets, list, context) => {
   const envList = list[plugin] || {};
   const filteredList = Object.keys(targets).reduce((a, b) => {
```
Comment (nicolo-ribaudo) [inline packages/babel-preset-env/src/debug.js]: Not related to this change, but `a` and `b` should really be renamed.

Comment (existentialism) [reply packages/babel-preset-env/src/debug.js]: Cleaned it up!

**Comment by brneto** [comment]
This is the right behavior for dotall-regex? The use of the unicode flag (`u`) should had no effect, shouldn't it?

Why `/hello.world/us.test(input)` is returning false?

**Test:**
```
const input = `
Lorem ipsum dolor sit amet, consectetur adipiscing hello
world elit. Nam sit amet elit id risus aliquam porta.
`;

describe('RegEx dotAll mode', () => {
  let received;
  it('should dot matches a line break', () => {
    received = /hello.world/.test(input);
    expect(received).toBe(false);

    received = /hello.world/u.test(input);
    expect(received).toBe(false);

    received = /hello.world/s.test(input);
    expect(received).toBe(true);

    received = /hello.world/us.test(input);
    expect(received).toBe(false); // Why the regex is false here?
  });
});
```
**package.json:**
   ```
    "@babel/polyfill": "^7.0.0-beta.49",
    "@babel/cli": "^7.0.0-beta.49",
    "@babel/core": "^7.0.0-beta.49",
    "@babel/node": "^7.0.0-beta.49",
    "@babel/plugin-proposal-class-properties": "^7.0.0-beta.49",
    "@babel/plugin-proposal-decorators": "^7.0.0-beta.49",
    "@babel/plugin-proposal-function-bind": "^7.0.0-beta.49",
    "@babel/plugin-proposal-object-rest-spread": "^7.0.0-beta.49",
    "@babel/plugin-proposal-optional-chaining": "^7.0.0-beta.49",
    "@babel/plugin-syntax-dynamic-import": "^7.0.0-beta.49",
    "@babel/preset-env": "^7.0.0-beta.49",
    "@babel/preset-react": "^7.0.0-beta.49",
    "babel-core": "^7.0.0-0",
    "babel-eslint": "8.2.1",
    "babel-jest": "^23.0.1",
    "babel-loader": "^8.0.0-beta.3",
    "babel-plugin-dynamic-import-node": "^1.2.0",
    "eslint-plugin-babel": "^5.1.0",
```
**.babelrc:**
```
{
  "presets": [
    ["@babel/preset-env", {
      "targets": { "node": "current" }
    }]
  ],
  "plugins": [
    ["@babel/plugin-proposal-decorators", { "legacy": true }],
    ["@babel/plugin-proposal-class-properties", { "loose": true }],
    "@babel/plugin-proposal-object-rest-spread",
    "@babel/plugin-proposal-function-bind",
    "@babel/plugin-proposal-optional-chaining",
    "dynamic-import-node"
  ]
}
```

**Comment by mathiasbynens** [comment]
That should work indeed. https://new.babeljs.io/docs/en/next/babel-plugin-transform-dotall-regex.html#example

**Comment by brneto** [comment]
My test behavior has to do with the issue #6931? The test `/hello.world/us.test(input)` should be `true`, right?

**Comment by mathiasbynens** [comment]
`/hello.world/us.test('hello\nworld')` should be `true`.

**Comment by brneto** [comment]
I've just transpiled (with babel) a small test code and look what I got:

**In:**
```
const r1 = /./s;
const r2 = /./su;
```

**Out:**
```
"use strict";

const r1 = /./s;
const r2 = /[\0-\t\x0B\f\x0E-\u2027\u202A-\u{10FFFF}]/su;
```

**Comment by mathiasbynens** [comment]
Looks like it’s not transpiling the `s` flag at all? Is https://github.com/babel/babel/tree/master/packages/babel-plugin-transform-dotall-regex not enabled somehow?

**Comment by brneto** [comment]
I'm using `@babel/preset-env` I think it is enabled by default. Should I take some additional action to get `dotall-regex` working?

**Comment by nicolo-ribaudo** [comment]
Which node version are you using? Maybe it already supports the dot-all flag.

**Comment by brneto** [comment]
node v9.11.1

**Comment by mathiasbynens** [comment]
There is indeed a bug here. `/./su` and `/[\0-\t\x0B\f\x0E-\u2027\u202A-\u{10FFFF}]/su` are not equivalent.

I’ve verified that regexpu-core handles this correctly:

```js
rewritePattern('.', 'su', { dotAllFlag: true, useUnicodeFlag: true });
// → '[\\0-\\u{10FFFF}]'
```

So it seems like in this case, Babel isn’t passing those flags to regexpu-core.

**Comment by mathiasbynens** [comment]
Looking at https://github.com/babel/babel/blob/9f4e2f81d8c1780ba435e880479427f45213421a/packages/babel-plugin-transform-unicode-regex/src/index.js and 
https://github.com/babel/babel/blob/9f4e2f81d8c1780ba435e880479427f45213421a/packages/babel-plugin-transform-dotall-regex/src/index.js, the problem must be that Babel applies `transform-unicode-regex` before it applies `transform-dotall-regex`.

`transform-unicode-regex` transpiles the pattern, including any `.` in the pattern, and removes the `u` flag. Then later, when the transpiled regular expression is passed to `transform-dotall-regex`, it never sees a regular expression containing a `.` anymore.

Can we swap the order in which the transforms get applied? That would fix the bug.

**Comment by existentialism** [comment]
Good ol' plugin ordering, created https://github.com/babel/babel/pull/8138

**Comment by brneto** [comment]
I think would be a good ideia to include a test like the one I’ve done above, don’t you think?

**Comment by existentialism** [comment]
@brneto thanks for reporting the issue, #8138 will be out in the next release!

**Pull request merged**